fbpic/particles/utilities/utility_methods.py

```diff
@@ -78,8 +78,8 @@ def weights(x, invdx, offset, Nx, direction, shape_order, beta_n):
             S[1,:] = 1 - S[0,:]
         # Linear weight r
         elif direction == 'r':
-            ir = np.minimum(np.maximum(i[0,:], 0), Nx-1)
-            S[0,:] = (i[1,:] - x_cell) * (1.+beta_n[ir]*( x_cell - i[0,:] ))
```

Comment:
Note that the above code means that particles located in the first half of the first cell were using a non-zero Ruyten coefficient (`beta_n[0]`, before a zero-element was added to the `beta_n` array)